crates/shared/src/signature_validator.rs

```diff
@@ -39,6 +50,13 @@ pub trait SignatureValidating: Send + Sync {
         &self,
         checks: Vec<SignatureCheck>,
     ) -> Vec<Result<(), SignatureValidationError>>;
+
+    /// Validates the signature and returns the `eth_estimateGas` of the
+    /// isValidSignature call minus the tx initation gas amount of 21k.
+    async fn validate_signature_and_get_additional_gas(
```

Comment:
We added this function to extend `validate_signature`.
I was a bit surprised we didn't also need to extend `validate_signatureS`: I checked and it appears to be unused.
Doesn't need to be in this PR, but I suppose we want to remove it.

Reply:
I am not sure I understand. 

`validate_signatures` is used by the autopilot to check before each run loop that the signature verification goes through. We decided to not check again the gas amounts, as a simplification. But in theory, an EIP order's isValidSignature could "charge little gas" during order creation and then later charge more.
But we said that this would be okay for the time being.